client/modules/Registration/Registration.js

```diff
@@ -3,33 +3,113 @@ import React, { Component, PropTypes } from 'react';
+  }
+
 export class Registration extends Component {
 
+    constructor() {
+        super();
+        this.state = {
+          email: '',
+          password1: '',
+          password2: '',
+          mobile: '',
+          license: '',
+          
+          everFocusedEmail: false,
+          everFocusedPassword1: false,
+          everFocusedPassword2: false,
+          everFocusedMobile: false,
+          everFocusedLicense: false,
+          inFocus: '',
+        };
+      }
+      
+      handleEmailChange = (evt) => {
+        this.setState({ email: evt.target.value });
+      }
+      
+      handlePassword1Change = (evt) => {
+        this.setState({ password1: evt.target.value });
+      }
+
+      handlePassword2Change = (evt) => {
+        this.setState({ password2: evt.target.value });
+      }
+
+      handleLicenseChange = (evt) => {
+        this.setState({ license: evt.target.value });
+      }
+
+      handleMobileChange = (evt) => {
+        this.setState({ mobile: evt.target.value });
+      }
+      
+      handleSubmit = (evt) => {
+        if (!this.canBeSubmitted()) {
+          evt.preventDefault();
+          return;
+        }
+        const { email, password, mobile, license } = this.state;
+        //alert(`Signed up with email: ${email} password: ${password1} mobile: ${mobile} license: ${license}`);
+      }
+      
+      canBeSubmitted() {
+        const errors = validate(this.state.email, this.state.password1, this.state.password2, this.state.mobile, this.state.license);
+        const isDisabled = Object.keys(errors).some(x => errors[x]);
+        return !isDisabled;
+      }
+
   render() {
-    // Here goes our page 
+    
+    const errors = validate(this.state.email, this.state.password1, this.state.password2, this.state.mobile, this.state.license);
+    const isDisabled = Object.keys(errors).some(x => errors[x]);
+
+    
     return (
         <div className={styles.body}>
             <h1 className={styles.title}>Registration</h1>
             <p><strong>Registration Form</strong><br /></p>
-            <form>
-                <label className={styles.labels} for="firstName">
-                    <div className={styles.labelText}>First Name *</div>
-                    <input type="text" name="firstName" id="firstName" />
-                </label>
-                <label className={styles.labels} for="lastName">
-                    <div className={styles.labelText}>Last name *</div>
-                    <input type="text" name="lastName" id="lastName"/>
-                </label>
-                <label className={styles.labels} for="emailAddr">
+            <form onSubmit={this.handleSubmit}>
+                <label className={styles.labels} for="email">
                     <div className={styles.labelText}>Email *</div>
-                    <input type="text" name="emailAddr" id="emailAddr"/>
+                    <input className={errors.email ? "error" : ""} type="text" placeholder="Enter Email"
```

Comment:
should `{errors.email ? "error" : ""}`
actually be `{errors.email ? style.error : ""}` ? 
i ran it locally and entered 'foo' and there was no update to the border of the input...

Reply:
I couldn't get it to work either way for some reason